### DISCUSSION TITLE
How to find which supported devices are routers?

### DISCUSSION THREAD

**Discussion by mfalkvidd** [Koenkk/zigbee2mqtt#13660] [Devices]
Hi.

Is there a way to find which of the [supported devices](https://www.zigbee2mqtt.io/supported-devices/) are Zigbee routers?

If this is currently not possible, would a new `Exposes` entry be feasible? Or some other way?

**Comment by DubhAd** [comment]
The _Device type_ tells you if it's a router, check the _About_ and _State_ pages in your Z2M UI.

Beyond that the rule of thumb is that any mains powered device will be a router _except_:
- Sengled bulbs
- Any two-wire switch or dimmer

**Reply by mfalkvidd** [reply]
Sorry, I don't see a way to filter the list of supported devices on _Device type_. I only see _Vendor_ and _Exposes_. I don't see a way to filter on whether the device is mains powered either.

**Reply by ncc0624** [reply]
Settings -> About -> Stats, 2. By device type

**Reply by mfalkvidd** [reply]
I don't see a settings link at https://www.zigbee2mqtt.io/supported-devices/

**Reply by ncc0624** [reply]
My mistake. I assumed you were referring to which devices you had that were routers. In that case best to follow the advice from @DubhAd generally all main electrical powered devices.

**Reply by mfalkvidd** [reply]
Awesome, thanks! Is there a way to filter on whether the device is mains powered? I tried searching for mains, but that only shows a single device.

**Reply by mfalkvidd** [reply]
I found this list https://zigbee.blakadder.com/routers.html
Not sure how it is generated and maintained, and it does not contain all routers. But it's better than nothing :)
